ios/Classes/handlers/PushHandlers.swift

```diff
@@ -9,18 +9,29 @@ public class PushHandlers: NSObject {
         if PushActivationEventHandlers.getInstance(methodChannel: ably.channel).flutterResultForActivate != nil {
             returnMethodAlreadyRunningError(result: result, methodName: "activate")
         } else if let push = getPush(instanceStore: ably.instanceStore, call: call, result: result) {
-            PushActivationEventHandlers.getInstance(methodChannel: ably.channel).flutterResultForActivate = result
-            push.activate()
+            #if targetEnvironment(simulator)
+                result(FlutterError(code: "PushHandlers_activate", message: "Device activation for push notifications does not work on iOS simulator because APNs is not supported.", details: nil))
+                return
+            #else
```

Comment:
te solution is OK, but  `return` above is redundant

Reply:
I know it's redundant _now_, but maybe I'm afraid that code gets added below the else that this code will be missed, and then this code . Premature optimization, so i fixed in: c87076b332c81ac369f62bbf750c2ab326f5eb6e

Reply:
I understand, so you can leave `return` but you should remove `#else` then :D

Reply:
I've removed all the returns in a new commit `9b89a44eb8d9d04ad80c72f2f41280301be4effd`. After thinking about it, I prefer this way since if/else is clearly on the same indentation level.